static/app/views/issueDetails/actions/newIssueExperienceButton.tsx

```diff
@@ -25,17 +26,100 @@ import {
 } from 'sentry/views/issueDetails/issueDetailsTourModal';
 import {useHasStreamlinedUI} from 'sentry/views/issueDetails/utils';
 
+/**
+ * This hook will cause the promotional modal to appear if:
+ *  - All the steps have been registered
+ *  - The tour has not been completed
+ *  - The tour is not currently active
+ *  - The streamline UI is enabled
+ *  - The user's browser has not stored that they've seen the promo
+ *
+ * Returns a function that can be used to reset the modal.
+ */
+export function useIssueDetailsPromoModal() {
+  const organization = useOrganization();
+  const hasStreamlinedUI = useHasStreamlinedUI();
+  const {mutate: mutateAssistant} = useMutateAssistant();
+  const {
+    startTour,
+    endTour,
+    currentStepId,
+    isRegistered: isTourRegistered,
+    isCompleted: isTourCompleted,
+  } = useIssueDetailsTour();
+
+  const [localTourState, setLocalTourState] = useLocalStorageState(
```

Comment:
If localstorage is actually necessary it seems like we would want to use it as a backup everywhere we currently use /assistant/ - do you think that's the case?

Reply:
I think maybe, though there are a bunch of other places we use the `assistant` that would need addressing so it's probably out of scope for now. If we have time we can look into the cause for this user's bad assistant state but I think it's a little early for a broad change (though it's worth noting)

If reports keep coming in we should look into it further, this page is just extremely high traffic so I'm being cautious.

Reply:
Also something we should note for building a `TourStore` to resolve the concurrent tour issues (related to https://github.com/getsentry/sentry/pull/89733)!